tests/components/auth/test_init.py

```diff
@@ -620,6 +624,67 @@ async def test_ws_delete_all_refresh_tokens(
         assert refresh_token is None
 
 
+async def test_ws_delete_all_refresh_tokens_filtered(
+    hass: HomeAssistant,
+    hass_admin_user: MockUser,
+    hass_admin_credential: Credentials,
+    hass_ws_client: WebSocketGenerator,
+    hass_access_token: str,
+) -> None:
+    """Test deleting all refresh tokens of a specific type."""
+    assert await async_setup_component(hass, "auth", {"http": {}})
+
+    # one token already exists
+    await hass.auth.async_create_refresh_token(
+        hass_admin_user, CLIENT_ID, credential=hass_admin_credential
+    )
+
+    # create a long lived token
+    await hass.auth.async_create_refresh_token(
+        hass_admin_user,
+        CLIENT_ID + "_LL",
+        client_name="client_ll",
+        credential=hass_admin_credential,
+        token_type=TOKEN_TYPE_LONG_LIVED_ACCESS_TOKEN,
+    )
+
+    await hass.auth.async_create_refresh_token(
+        hass_admin_user, CLIENT_ID + "_1", credential=hass_admin_credential
+    )
+
+    ws_client = await hass_ws_client(hass, hass_access_token)
+
+    # get all tokens
+    await ws_client.send_json({"id": 5, "type": "auth/refresh_tokens"})
+    result = await ws_client.receive_json()
+    assert result["success"], result
+
+    tokens = result["result"]
+
+    await ws_client.send_json(
+        {
+            "id": 6,
+            "type": "auth/delete_all_refresh_tokens",
+            "token_type": TOKEN_TYPE_NORMAL,
+        }
+    )
+
+    result = await ws_client.receive_json()
+    assert result, result["success"]
+    remaining_tokens = 0
+    for token in tokens:
+        refresh_token = await hass.auth.async_get_refresh_token(token["id"])
+        if refresh_token is not None:
+            remaining_tokens += 1
+
+        if token["type"] == TOKEN_TYPE_NORMAL:
+            assert refresh_token is None
+        else:
+            assert refresh_token.client_id == token["client_id"]
```

Comment:
Usually we try to avoid branching in tests and use `pytest.mark.parametrize` instead

Reply:
done (I think I followed what you were asking for, but I might not be 100% sure).